### PR TITLE
Fix #3883. Fix #3884. Somatic score display and filter.

### LINKED ISSUE #3883: add `SOMATICSCORE` for SVs
The `VCF` for SVs contains `SOMATICSCORE` computed by manta for tumor-normal cases. It would be nice to have it as a list in the structural variants view in scout.

### LINKED ISSUE #3884: Filter by `SOMATICSCORE`
It will be even great to have the possibility to filter SVs on the `SOMATICSCORE` as described and requested in https://github.com/Clinical-Genomics/scout/issues/3883

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality.

Somatic Structural variantS page. We filter away low scoring variants, keeping non existing calls to leave variants only found by callers other than e.g. manta that have it.

Screenshot 2023-05-31 at 11 53 09

Stuctural variant page:

Screenshot 2023-05-31 at 11 33 42

As this anyway touches the somatic structural variantS view, why not add callers as well. Just display for now, as in issue.

Screenshot 2023-05-31 at 13 57 40

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by CR
- [x] tests executed by DN

### CODE REVIEW

File: scout/adapter/mongo/query.py [hunk 1]
```diff
@@ -717,6 +717,16 @@ def secondary_query(self, query, mongo_query, secondary_filter=None):
             if criterion == "alt_count":
                 mongo_secondary_query.append({"tumor.alt_depth": {"$gt": query.get("alt_count")}})
 
+            if criterion == "somatic_score":
+                mongo_secondary_query.append(
+                    {
+                        "$or": [
+                            {"somatic_score": {"$gt": query.get("somatic_score")}},
+                            {"somatic_score": {"$exists": False}},
```
Comment (northwestwitch) [inline scout/adapter/mongo/query.py:725]: Bit this way you return also those without somatic score, shouldn't this be True instead? 

So if you have a case without somatic score and you filter using the score then you don't get any variants, but if the case has somatic score then you get only variants with scopre higher than the threshold

**Review by northwestwitch — COMMENTED** [review]
I have some comments/questions

File: scout/adapter/mongo/query.py (hunk 1 above)
Comment (dnil) [reply scout/adapter/mongo/query.py:725]: Why would you want that?

File: scout/server/blueprints/variants/forms.py [hunk 2]
```diff
@@ -201,6 +201,7 @@ class CancerSvFiltersForm(SvFiltersForm):
 
     depth = IntegerField("Depth >", validators=[validators.Optional()])
     alt_count = IntegerField("Min alt count", validators=[validators.Optional()])
+    somatic_score = IntegerField("Somatic score", validators=[validators.Optional()])
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/variants/forms.py:204]: > and if they are not completely obvious to the user,

I'm sure, the question comes from a non-user in fact 😄

File: scout/adapter/mongo/query.py (hunk 1 above)
Comment (dnil) [reply scout/adapter/mongo/query.py:725]: From the text above: "We filter away low scoring variants, keeping non existing calls to leave variants only found by callers other than e.g. manta that have it.". At least to me, this is the way. Compare GnomAD etc. We very rarely want to remove variants from triage on missing indication, only if we have some kind of clear indication like high observed frequency, high observed CADD etc. Or a low observed somatic score. But not no-call.

Comment (northwestwitch) [reply scout/adapter/mongo/query.py:725]: I mean if you want those with a score higher than a number, shouldn't it be

```
"$and": [
                            {"somatic_score": {"$gt": query.get("somatic_score")}},
                            {"somatic_score": {"$exists": True}},
```

Comment (dnil) [reply scout/adapter/mongo/query.py:725]: Nah, firstly gt kind of implies exists, secondly then the feature could be used to exclude calls from callers that dont implement somatic_score. That would not be very neat for the clinical side, although I'm sure some researchers would be happy.

Comment (northwestwitch) [reply scout/adapter/mongo/query.py:725]: I don't know, to me it feels weird to filter by somatic score and get variants that have no somatic score at all..

Comment (dnil) [reply scout/adapter/mongo/query.py:725]: Huh, compare gnomAD, loqus counts, swegen, CADD?

Comment (northwestwitch) [reply scout/adapter/mongo/query.py:725]: But I understand that you might want the filtering to be conservative. In this case perhaps explain in the tooltip that returned variants have score higher than filter or no score at all?

Comment (northwestwitch) [reply scout/adapter/mongo/query.py:725]: > Huh, compare gnomAD, loqus counts, swegen, CADD?

I know, but they are also a bit weird for me :)

File: scout/server/blueprints/variants/forms.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dnil) [reply scout/server/blueprints/variants/forms.py:204]: This is a returning issue especially when showing Scout to a new researcher. I like that all the filters are simplified to as much possible to just do the Right Thing when used clinically, but we probably in hindsight should have added a research query/filter page as well. That would have kept things so much smoother during devs. Its not too late.. 😁

**Review by northwestwitch — APPROVED** [review]
Looks good! 👍🏻

File: scout/server/blueprints/variants/templates/variants/cancer-sv-variants.html [hunk 3]
```diff
@@ -121,7 +121,7 @@
     <td>
       {{ variant_funct_anno_cell(variant) }}
     </td>
-    <td>{{ allele_cell(variant.tumor or {}) }}</td>
+    <td>{{ allele_cell(variant.tumor or {}) }}{% if variant.somatic_score %}<small class="text-muted">[{{ variant.somatic_score }}]</small>{% endif %}</td>
```
[the remark this answers is not in the document]
Comment (dnil) [reply scout/server/blueprints/variants/templates/variants/cancer-sv-variants.html:126]: Ah, I didn't see those! The screen size... 😄 Hm, let me try playing a little with the witdth% - I worry the rows will be too thick.

File: scout/server/blueprints/variants/forms.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/variants/forms.py:204]: Give me a month and this will be a returning issue for me as well (I can't remember anything!!!) 😆

**Comment by dnil** [comment]
Thanks a bunch for the quick review! I played with the witdths to make it look nice down to at least say 1400px. Honestly a lot of things (in Scout) start looking weird to me at <1600px. But I guess it is occasionally used at much less. I wonder how it went with that screen size poll/data collection?

**Comment by northwestwitch** [comment]
> I wonder how it went with that screen size poll/data collection?

I have no idea!

**Comment by dnil** [comment]
Looks good on stage methinks.
Screenshot 2023-05-31 at 16 32 00

**Pull request merged**